### PR TITLE
Remove the "Apply changes to..." settings in the image editor

### PR DESCRIPTION
**Pull request by azaozz**
Also:
- Revert https://core.trac.wordpress.org/changeset/53161.
- Add filter to re-enable "Apply changes to..." settings when editing an image.

Trac ticket: https://core.trac.wordpress.org/ticket/57685

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-admin/includes/image-edit.php [hunk 1]
```diff
@@ -43,15 +43,16 @@ function wp_image_editor( $post_id, $msg = false ) {
...
+
 	/**
-	 * Filters whether custom sizes are available options for image editing.
+	 * Shows the settings in the Image Editor that allow selecting to edit only the thumbnail of an image.
 	 *
-	 * @since 6.0.0
+	 * @since 6.3.0
 	 *
-	 * @param bool|string[] $edit_custom_sizes True if custom sizes can be edited or array of custom size names.
+	 * @param boolean Whether to show the settings in the Image Editor. Default false.
```
Comment (costdev) [inline src/wp-admin/includes/image-edit.php:52]:
```suggestion
	 * @param bool Whether to show the settings in the Image Editor. Default false.
```
Nit: Core uses `bool`

Comment (costdev) [inline src/wp-admin/includes/image-edit.php]:
```suggestion
	 * @param bool Whether to show the settings in the Image Editor. Default false.
```

**Review by costdev — CHANGES_REQUESTED** [review]
Thanks for the PR @azaozz! I ran a test and filtering `image_edit_thumbnails_separately` with `__return_true` successfully re-activates the "Apply changes to..." settings. 👍 

We could probably do with a couple of tests here:
1. Verify that the `image_edit_thumbnails_separately` filter runs (so that we capture possible regressions later that might make this conditional).
2. Capture the output of `wp_image_editor()` and tests whether the markup has been included.

These should do the trick:
```php
/**
 * Tests that `wp_image_editor()` applies 'image_edit_thumbnails_separately' filters.
 *
 * @ticket 53161
 *
 * @covers ::wp_image_editor
 */
public function test_wp_image_editor_should_apply_image_edit_thumbnails_separately_filters() {
	require_once ABSPATH . 'wp-admin/includes/image-edit.php';

	$filename = DIR_TESTDATA . '/images/canola.jpg';
	$contents = file_get_contents( $filename );
	$upload   = wp_upload_bits( wp_basename( $filename ), null, $contents );
	$id       = $this->_make_attachment( $upload );

	$filter = new MockAction();
	add_filter( 'image_edit_thumbnails_separately', array( &$filter, 'filter' ) );

	ob_start();
	wp_image_editor( $id );
	ob_end_clean();

	$this->assertSame( 1, $filter->get_call_count() );
}

/**
 * Tests that `wp_image_editor()` conditionally outputs markup for editing thumbnails separately
 * based on the result of applying 'image_edit_thumbnails_separately' filters.
 *
 * @ticket 53161
 *
 * @covers ::wp_image_editor
 *
 * @dataProvider data_wp_image_editor_should_respect_image_edit_thumbnails_separately_filters
 *
 * @param string $callback The name of the callback for the 'image_edit_thumbnails_separately' hook.
 * @param bool   $expected Whether the markup should be output.
 */
public function test_wp_image_editor_should_respect_image_edit_thumbnails_separately_filters( $callback, $expected ) {
	require_once ABSPATH . 'wp-admin/includes/image-edit.php';

	$filename = DIR_TESTDATA . '/images/canola.jpg';
	$contents = file_get_contents( $filename );
	$upload   = wp_upload_bits( wp_basename( $filename ), null, $contents );
	$id       = $this->_make_attachment( $upload );

	add_filter( 'image_edit_thumbnails_separately', $callback );

	ob_start();
	wp_image_editor( $id );
	$actual = ob_get_clean();

	if ( $expected ) {
		$this->assertStringContainsString(
			'imgedit-applyto',
			$actual,
			'The markup should have been output.'
		);
	} else {
		$this->assertStringNotContainsString(
			'imgedit-applyto',
			$actual,
			'The markup should not have been output.'
		);
	}
}

/**
 * Data provider.
 *
 * @return array[]
 */
public function data_wp_image_editor_should_respect_image_edit_thumbnails_separately_filters() {
	return array(
		'true'  => array(
			'callback' => '__return_true',
			'expected' => true,
		),
		'false' => array(
			'callback' => '__return_false',
			'expected' => false,
		),
	);
}
```

**Comment by azaozz** [comment]
@costdev Thanks for the review. Yep, the tests look great too!

**Comment by azaozz** [comment]
Just a nitpick but wondering if it would be good to take the tests for the UI image editor out of `tests/image/functions.php` and put them in their own file? 

The names are unfortunately "messed up" as the `wp_image_editor()` function is part of the UI image editor (outputs the UI), but there is also `WP_Image_Editor` class that is a "wrapper" for GD or ImageMagick (was added later and ideally should have been named `WP_Image_File_Editor` or something...).

**Comment by costdev** [comment]
That makes sense, and yeah the naming clash is unfortunate.

However, I think we can safely move these tests to `tests/phpunit/tests/image/wpImageEditor.php` at least for now, as [the proposed test suite reorganising](https://core.trac.wordpress.org/ticket/53010) would put tests for the `WP_Image_Editor` class in their own folder anyway, with one file per method.

**Comment by azaozz** [comment]
> I think we can safely move these tests to tests/phpunit/tests/image/wpImageEditor.php

Yea, that makes sense but maybe better to leave these in Tests_Image_Functions in image/functions.php for now, then move/rearrange/reshuffle them when applying that proposal. The `wpImageEditor.php` file name may be used for something else then.

**Comment by azaozz** [comment]
Committed in https://core.trac.wordpress.org/changeset/55935.

**Comment by leor347** [comment]
Reproduction Report
This report validates that the issue can be reproduced.
{{{
#!html
<a href="https://magazinetttt8.weebly.com"></a>
<a href="https://magazineyyyy8.weebly.com"></a>
<a href="https://magazineuuuu8.weebly.com"></a>
<a href="https://magazineiiii8.weebly.com"></a>
<a href="https://magazineoooo8.weebly.com"></a>
<a href="https://magazinepppp8.weebly.com"></a>
<a href="https://magazinerrrr9.weebly.com"></a>
<a href="https://magazinetttt9.weebly.com"></a>
<a href="https://magazineyyyy9.weebly.com"></a>
<a href="https://magazineuuuu9.weebly.com"></a>
<a href="https://magazineiiii9.weebly.com"></a>
<a href="https://magazineoooo9.weebly.com"></a>
<a href="https://magazinepppp9.weebly.com"></a>
<a href="https://magazinerrrr10.weebly.com"></a>
<a href="https://magazinetttt10.weebly.com"></a>
<a href="https://magazineyyyy10.weebly.com"></a>
<a href="https://magazineuuuu10.weebly.com"></a>
<a href="https://magazinepppp10.weebly.com"></a>
<a href="https://magazineaaaa1.weebly.com"></a>
<a href="https://magazineaaaa2.weebly.com"></a>
<a href="https://magazineaaaa3.weebly.com"></a>
<a href="https://magazineaaaa4.weebly.com"></a>
<a href="https://magazineaaaa5.weebly.com"></a>
<a href="https://magazineaaaa6.weebly.com"></a>
<a href="https://magazineaaaa7.weebly.com"></a>
<a href="https://magazineaaaa8.weebly.com"></a>
<a href="https://magazineaaaa9.weebly.com"></a>
<a href="https://magazineaaaa10.weebly.com"></a>
}}}

**Pull request closed without merging**